fire/core.py

```diff
@@ -92,9 +92,14 @@ def Fire(component=None, command=None, name=None):
     it's a class). When all arguments are consumed and there's no function left
     to call or class left to instantiate, the resulting current component is
     the final result.
-    If the trace command line argument is supplied, the FireTrace is returned.
+    If a Fire error is encountered, the Fire Trace is displayed to stdout and
+    a FireExit is raised.
+    If the trace command line argument is supplied, the FireTrace is returned
```

Comment:
@dbieber - If you pass `--trace` and also have invalid params, should it exit 0 or non-zero, I wasn't sure?